jsk_vil_ros/apis/ofa/server.py

```diff
@@ -0,0 +1,278 @@
+# sys
+import os
+# inference
+import torch
+import numpy as np
+import cv2
+from fairseq import utils, tasks
+from fairseq import checkpoint_utils
+from fairseq import options
+from fairseq.dataclass.utils import convert_namespace_to_omegaconf
+from utils.eval_utils import eval_step
+from utils.zero_shot_utils import zero_shot_step
+from tasks.mm_tasks.caption import CaptionTask
+from tasks.mm_tasks.refcoco import RefcocoTask
+from tasks.mm_tasks.vqa_gen import VqaGenTask
+from models.ofa import OFAModel
+from torchvision import transforms
+from PIL import Image
+# web server
+from flask import Flask, request, Response
+import json
+import base64
+
+PARAM_DIR = "/var/mount/params"
+# OFA_PARAM[TASK][SCALE]
+OFA_PARAM = {
+    "caption":{
+        "large":"caption_large_best_clean.pt",
+        "huge":"caption_huge_best.pt"
+    },
+    "refcoco":{
+        "large":"refcocog_large_best.pt",
+        "huge":"refcocog_large_best.pt"
+    },
+    "vqa_gen":{
+        "large":"vqa_large_best.pt",
+        "huge":"vqa_large_best.pt"
+    }
+}
+
+def apply_half(t):
+    if t.dtype is torch.float32:
+        return t.to(dtype=torch.half)
+    return t
+
+class Inference:
```

Comment:
```suggestion
class Inference(object):
```

Reply:
class name should be `InferenceServer` or `HogehogeServer`.
and the program name should be `inference_server.py` or `hogehoge_server.py`